doc/specs/#5000 - Process Model 2.0/#5000 - Process Model 2.0.md

```diff
@@ -0,0 +1,1253 @@
+     building.
+- Doing this after 1 might be helpful.
+- Needs 14 to be able to specify the location of the new window.
+
+<hr>
+
+16. Add support for a `NewWindow` action
+17. `wt` accepts a commandline arg to force it to auto-elevate itself if it
+    isn't already elevated.
+18. (Dependent on 16, 17) Users can mark a profile as `"elevated": true`, to
+    always force a new elevated window to be created for elevated profiles.
+19. (Dependent on 18, 13) When a user creates an elevated profile, we'll
+    automatically create an elevated version of the current window, and move all
+    the current tabs to that new window.
+
+## Footnotes
+
+<a name="footnote-1"><a>[1]: See [Quake mode scenarios] for a longer enumeration
+of possible scenarios.
+
+<a name="footnote-2"><a>[2]: I'm not prescribing that this solution _must_
+involve JSON. In this scenario, I'm using JSON merely as an example
+serialization that we can quickly examine and discuss. If we did use JSON, we'd
+almost certainly want to use a string as the payload type, so we can just pass
+that string into some WinRT method projected by whatever type corresponds to the
+provided `type`.
+
+<a name="footnote-3"><a>[3]: The state that's serialized here for the contents
+of a window might also be convenient to re-use for the restoration of terminal
+window state across reboots. If we already know how to serialize the entire
+state of a Terminal window, then storing that somewhere for a future terminal
+launch to use seems like an obvious next step. See also [#961].
+
+## Future considerations
+
+* A previous discussion with someone on the input team brought up the following
+  idea: When a tab is being torn out, move all the _other_ tabs to a new window,
+  and continue dragging the current window. This should make the drag/drop
+  experience feel more seamless, and might be able to allow us to render the tab
+  content as we're drag/dropping.
+  - If we pursue this, then we'll need to make sure that we re-assign the window
+    ID's as appropriate. the _new_ window (with the tabs that are being left
+    behind) should still have the same peasant ID as the original window, which
+    will now get a new ID (as to )
+* We could definitely do a `NewWindowFromTab(index:int?)` action that creates a
+  new window from a current tab once this all lands.
+  - Or it could be `NewWindowFromTab(index:union(int,int[])?)` to specify the
+    current tab, a specific tab, or many tabs.
+* During
+  [review](https://github.com/microsoft/terminal/pull/7240#issuecomment-716022646),
+  it was mentioned that when links are opened in the new Edge browser, they will
+  only glom onto an existing window if that window is open in the current
+  virtual desktop. This seems like a good idea of a feature for the Terminal to
+  follow as well. Since Edge is able to do it, there must be some way for an
+  application to determine which virtual desktop it is open on. We could use

```

Comment:
[`IVirtualDesktopManager`](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ivirtualdesktopmanager) should do the trick!

As far as I know that's actually the extent of the public virtual desktop API in Windows. I've seen some other stuff used in [this project](https://github.com/Grabacr07/VirtualDesktop) (seemingly taken from [this Russian blog](https://www.cyberforum.ru/blogs/105416/blog3671.html)) for things like creating visual desktops and getting notifications when they're created, but all that is seemingly undocumented and not meant for public consumption. (Not that Terminal needs it anyway.)